prov/shm/src/smr_av.c

```diff
@@ -170,20 +170,22 @@ static int smr_av_lookup(struct fid_av *av, fi_addr_t fi_addr, void *addr,
 	name = smr_av->smr_map->peers[id].peer.name;
 
 	strncpy((char *) addr, name, *addrlen);
-
-	((char *) addr)[MIN(*addrlen - 1, strlen(name))] = '\0';
-	*addrlen = strlen(name) + 1;
+	*addrlen = MIN(*addrlen, strlen(name)+1);
+	if (*addrlen)
+		((char *) addr)[*addrlen - 1] = '\0';
 	return 0;
 }
 
 static const char *smr_av_straddr(struct fid_av *av, const void *addr,
 				  char *buf, size_t *len)
 {
 	/* the input address is a string format */
-	if (buf)
+	if (buf && *len) {
 		strncpy(buf, (const char *)addr, *len);
-
-	*len = strlen((const char *)addr) + 1;
+		*len = MIN(strlen((const char *)addr)+1, *len);
+		if (*len)
+			buf[*len - 1 ] = '\0';
+	}
```

Comment:
Please move the changes to the smr provider to its own patch.  Make sure the commit title indicates the provider.  E.g. prov/shm: Add checks for string overflow